src/core/server/rendering/rendering_service.tsx

```diff
@@ -132,11 +132,15 @@ export class RenderingService {
         externalUrl: http.externalUrl,
         vars: vars ?? {},
         uiPlugins: await Promise.all(
-          [...uiPlugins.public].map(async ([id, plugin]) => ({
-            id,
-            plugin,
-            config: await getUiConfig(uiPlugins, id),
-          }))
+          [...uiPlugins.public].map(async ([id, plugin]) => {
+            const { browserConfig, exposedConfigKeys } = await getUiConfig(uiPlugins, id);
+            return {
+              id,
+              plugin,
+              config: browserConfig,
+              exposedConfigKeys, // TODO: only include this for integration tests (what's the best way to do this?)
```

Comment:
I left this TODO here.

It's not the end of the world to expose these config keys to the browser, but it seems messy. I'd prefer to only do this during testing (or, only do this when running from source?)
I'm open to ideas.

Reply:
FTR tests are running in production mode, there isn't really a way to dissociate FTR and production (and we shouldn't). Eventually we could use a config setting, but that feels hacky.

As discussed sync (and to initiate the discussion), I think we could simply expose those keys only via a server API, and use a test plugin for FTR with an endpoint that would expose them

Reply:
I changed this in c553bc3d73b7009bef951b759ae4e4c02fa04d3f, let me know what you think of the approach.

Reply:
Not fond of the test-driven-feature methodology, but it will have to do.